### PR TITLE
fix(jenkins) refactor (apply code conventions) plugin to fix changelog output

### LINKED ISSUE #480: Jenkins Sources do not retrieve changelog
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When using a jenkins resource as "source", the changelog is not retrieved unless you specify additional github token + username.

If you don't, then the changelog is filed with a big warning text block:

```
Warning:
========
We currently rely on Github Release to get Jenkins changelog 
which requires a github token and a github username as in the following example:
---
source:
  kind: jenkins
  spec:
    release: stable
    github:
      token: {{ requiredEnv .github.token }}
      username: {{ .github.username }}
---
```

Jenkins changelogs for "human reading" are located on the following locations:

- Weekly: https://www.jenkins.io/changelog/
- Stable: https://www.jenkins.io/changelog-stable/

Thing is that the GitHub release changelogs for jenkins are not really the "human readable" changelogs:

- Stable changelogs are redirecting to the corresponding section (version as URL anchor) in https://www.jenkins.io/changelog-stable/
- Weekly is an automated generated changelog, raw from pull requests.

### Expected Behavior

Proposition is define the changelog returned by updatecli to point users to the jenkins.io webpage.

For instance:

```
CHANGELOG:
----------

Jenkins changelog is available at: https://www.jenkins.io/changelog-stable/#v2.319.2

```

### Steps To Reproduce

```yaml
title: "Bump Jenkins stable version"

sources:
  getLatestJenkinsStable:
    kind: jenkins
  getLatestJenkinsWeekly:
    kind: jenkins
    spec:
      release: weekly
```

### Environment

```markdown
- OS: Linux, macOS
- updatecli: 0.18.3
```

### Anything else?

This issue comes from the refactor work that I started on #412 .

It would allow removing a lot of code from the jenkins plugin (no need for github changelog retrieval)

### PR DESCRIPTION
**Pull request by dduportal**
Depends on #481 (need rebase once merged).

Fix #480.

This PR is a refactor of the jenkins plugin to allow continuing the work in #412.

It introduces the following changes:

* Update the jenkins example + add it to the end 2 end testing framework
* Apply code conventions for the plugin maven (exports a Spec object and a New(spec) constructor method
* Add unit test with the same method as for the maven plugin in #481 (hence the dependency)
* Remove unused code around the Jenkins "sem" version by using the `"github.com/Masterminds/semver/v3"` library, like in the `version` package
* Moved the meta URL in a constant as asked in https://github.com/updatecli/updatecli/pull/481/files#r790494074

## Test

To test this pull request, you can run the following commands:

```shell
go build -o ./dist/updatecli && ./dist/updatecli diff --config examples/updatecli.generic/jenkins.yaml
```

## Additional Information
 N.A.

### CODE REVIEW

**Comment by dduportal** [comment]
PR rebased since #481 has been merged.

For this PR, I'll wait for both your approval before merging, because it fixes a behavior :)

**Comment by dduportal** [comment]
Ping @olblak I would like to have this in 0.19.0

File: pkg/plugins/jenkins/changelog.go [hunk 1]
```diff
@@ -1,50 +1,33 @@
...
+	if release.ParsedVersion == "" {
+		return "", fmt.Errorf("Empty Jenkins version: %q", release.ParsedVersion)
 	}
-	changelog, err := g.Changelog(version.Version{
-		OriginalVersion: "jenkins-" + release.OriginalVersion,
-		ParsedVersion:   "jenkins-" + release.ParsedVersion})
 
-	return changelog, err
+	changelog := fmt.Sprintf(
+		"Jenkins changelog is available at: https://www.jenkins.io/%s/#v%s\n",
```
Comment (olblak) [inline pkg/plugins/jenkins/changelog.go:27]: I was going to suggest this approach :D

Comment (dduportal) [reply pkg/plugins/jenkins/changelog.go:27]: ;)

was suggested in https://github.com/updatecli/updatecli/issues/480

**Review by olblak — APPROVED** [review]
🥳

**Pull request merged**